src/Common/tests/Tests/System/StringTests.Span.cs

```diff
@@ -0,0 +1,1569 @@
+               "neutral, PublicKeyToken=b77a5c561934e089\u0022 version=\u00221\u0022 Unrestricte" +
+               "d=\u0022true\u0022/><IPermission class=\u0022System.Diagnostics.PerformanceCount" +
+               "erPermission, System, Version={VERSION}, Culture=neutral, PublicKeyToken=b77a5c5" +
+               "61934e089\u0022version=\u00221\u0022 Unrestricted=\u0022true\u0022/><IPermission" +
+               " class=\u0022System.Data.OleDb.OleDbPermission, System.Data, Version={VERSION}, " +
+               "Culture=neutral, PublicKeyToken=b77a5c561934e089\u0022 version=\u00221\u0022 Unr" +
+               "estricted=\u0022true\u0022/><IPermission class=\u0022System.Data.SqlClient.SqlCl" +
+               "ientPermission, System.Data, Version={VERSION}, Culture=neutral, PublicKeyToken=" +
+               "b77a5c561934e089\u0022 version=\u00221\u0022 Unrestricted=\u0022true\u0022/><IPe" +
+               "rmission class=\u0022System.Security.Permissions.DataProtectionPermission, Syste" +
+               "m.Security, Version={VERSION}, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a\u0022" +
+               " version=\u00221\u0022 Unrestricted=\u0022true\u0022/></PermissionSet><Permissio" +
+               "nSet class=\u0022System.Security.NamedPermissionSet\u0022version=\u00221\u0022 N" +
+               "ame=\u0022Nothing\u0022 Description=\u0022{Policy_PS_Nothing}\u0022/><Permission" +
+               "Set class=\u0022System.Security.NamedPermissionSet\u0022version=\u00221\u0022 Na" +
+               "me=\u0022Execution\u0022 Description=\u0022{Policy_PS_Execution}\u0022><Permissi" +
+               "on class=\u0022System.Security.Permissions.SecurityPermission, mscorlib, Version" +
+               "={VERSION}, Culture=neutral, PublicKeyToken=b77a5c561934e089\u0022version=\u0022" +
+               "1\u0022 Flags=\u0022Execution\u0022/></PermissionSet><PermissionSet class=\u0022" +
+               "System.Security.NamedPermissionSet\u0022version=\u00221\u0022 Name=\u0022SkipVer" +
+               "ification\u0022 Description=\u0022{Policy_PS_SkipVerification}\u0022><Permission" +
+               " class=\u0022System.Security.Permissions.SecurityPermission, mscorlib, Version={" +
+               "VERSION}, Culture=neutral, PublicKeyToken=b77a5c561934e089\u0022version=\u00221\u0022" +
+               " Flags=\u0022SkipVerification\u0022/></PermissionSet></NamedPermissionSets>";
+
+        [Theory]
+        [MemberData(nameof(GetEnumerator_TestData))]
+        public static void GetEnumerator(string s)
+        {
+            var enumerator = s.AsSpan().GetEnumerator();
+
+            int counter = 0;
+            while (enumerator.MoveNext())
+            {
+                Assert.Equal(s[counter], enumerator.Current);
+                counter++;
+            }
+            Assert.Equal(s.Length, counter);
+        }
+
+        [Fact]
+        public static void GetEnumerator_NonGeneric_Invalid()
```

Comment:
@ahsonkhan i added GetEnumerator tests.

Reply:
Can you please add such tests in a subsequent PR, separately? Otherwise, the code changes keep growing and become harder to review. Thanks.

Reply:
Yep